tinygrad/codegen/kernel.py

```diff
@@ -241,8 +241,6 @@ def simplify_ones(self) -> bool:
     # TODO: this should be factored in to multi shape stride
     if self.shape_len == 0: return False
     all_ones = [s==1 for s in self.full_shape]
-    self.local_dims -= sum(all_ones[self.first_reduce-self.local_dims:self.first_reduce])
```

Comment:
I don't think removing this is right

Reply:
This PR makes it so it's one way from globals to either LOCAL or UPCAST (see lines 411/429), so it's not possible.  It might not be the desired behavior though.

As discussed live, shelving this PR and reworking it into more testing and bug fixes before functional changes.